File: Labs and Projects/Week 6/sentimental-readability/readability.py
# TODO
import cs50


def count_letters(text):
    letterlength = 0
    for i in range(0, len(text)):
        if text[i].isalpha():
            letterlength += 1
    return letterlength


def count_words(text):
    wordlength = 0
    for i in range(0, len(text)):
        if text[i] == ' ':
            wordlength += 1
    return wordlength + 1


def count_sentences(text):
    sentencelength = 0
    for i in range(0, len(text)):
        if text[i] in ['.', '!', '?']:
            sentencelength += 1
    return sentencelength


text = cs50.get_string("Write Sample Text Here:")
letterlength = count_letters(text)
wordlength = count_words(text)
sentencelength = count_sentences(text)

L = letterlength / wordlength * 100
S = sentencelength / wordlength * 100

index = 0.0588 * L - 0.296 * S - 15.8

if (index >= 16):
    print("Grade 16+\n")
elif (index < 1):
    print("Before Grade 1 \n")
else:
    grade = round(index)
    print(f"Grade {grade}\n")

File: Labs and Projects/Week 2/readability/readability.c
u#include <cs50.h>
#include <stdio.h>
#include <math.h>

int count_letters(string text)
{

    int letterlength = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {

        //if character is upper of lower case alphabet cound as letter
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            letterlength ++;
        }

    }

    return letterlength;

}

int count_words(string text)
{

    int wordlength = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {

        //if character is space cound as word
        if (text[i] == ' ')
        {
            wordlength ++;
        }

    }

    //add 1 for final word
    return wordlength + 1;

}

int count_sentences(string text)
{

    int sentencelength = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {

        //if character is . ! or ? cound as sentence
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentencelength ++;
        }

    }

    return sentencelength;

}

int main(void)
{

    string text = get_string("Write sample text here:\n");

    //get lengths
    int letterlength = count_letters(text);
    int wordlength = count_words(text);
    int sentencelength = count_sentences(text);

    //calculate index
    float L = (float) letterlength / (float) wordlength * 100;
    float S = (float) sentencelength / (float) wordlength * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;


    //format output
    if (index >= 16)
    {

        printf("Grade 16+\n");

    }
    else if (index < 1)
    {

        printf("Before Grade 1\n");

    }
    else
    {

        int grade = round(index);
        printf("Grade %d\n", grade);

    }
}